### ISSUE TITLE
iOS: message list doesn't respect system text size

### ISSUE THREAD

**Issue by gnprice** [zulip/zulip-mobile#4042]
This is just like #2559, but I saw it just now in a recent version from master. This was on a simulated iPhone 8 running iOS 13.3.

> I had the Zulip app open to the message list screen, then went over to system settings and changed the text size. The text in the app bar responded appropriately, but the message list didn't change.
>
> This would make most of the app's functionality totally unusable for someone with a visual impairment who needs a large text size.

Not sure what changed between 642b2d43b, where I previously fixed this bug, and now. The only change since then that looks at all related is the very next commit, cbf2e6ce9, which deleted the `font-size:` line entirely.

Or maybe there's some other variable involved? I.e. I'm just testing it differently than I did when I wrote 642b2d43b, and the old way would still work fine now while the new way would already have reproed back then.

**Comment by gnprice** [comment]
> The only change since then that looks at all related is the very next commit, [cbf2e6c](https://github.com/zulip/zulip-mobile/commit/cbf2e6ce94ca4d9f3744d59030ff1671dd690cac), which deleted the `font-size:` line entirely.

Just to be sure, I tested just now the hypothesis that this somehow caused the bug: in Safari's Web Inspector, I added a `font-size: 93.75%` attribute back to the `html` CSS block, just like the one that commit removed. The bug still reproed -- the font just got slightly smaller, as you'd expect.

**Comment by gnprice** [comment]
The bug is already present in 26.21.144, which is the oldest release that Apple still keeps around in TestFlight even for me as the app's maintainer to install for testing. That's on my iPhone 7 running iOS 12.4.1; I've also confirmed the bug is present on that version in the latest release.

The release that originally had (what I sure thought at the time was) the fix was 25.3.117.

It would be great for someone to take this on, and debug and fix it. The next steps I'd pursue are:
* Try to find an older version of the app where this works (where the bug wasn't there.) In particular, confirm it worked in 25.3.117, or at commit 642b2d43b.
  * If you run into trouble running older versions of the app in development, see our [troubleshooting steps](https://github.com/zulip/zulip-mobile/blob/master/docs/howto/build-run.md). If you run into anything that isn't there, please describe it in chat -- debugging any issues of that kind will be valuable in itself.
* If it doesn't work at the commit that was supposed to fix it, that'll be interesting and we'll have to debug further from there.
* Alternatively once you do find a commit where it works, the next step is to "bisect" to narrow down and find the first commit that broke it. If this isn't familiar, ask in chat; it's a powerful tool which I'll be glad to explain.
* Then hopefully that will tell us a lot about what might be causing it!

**Comment by jainkuniya** [comment]
I was investigating on this:
- still have to figure out, when this was working and from where it broke. Facing issue with iOS build https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/ios.20build.20fail
- got his article https://www.interactiveaccessibility.com/blog/text-resizing-web-pages-ios-using-dynamic-type#.XqQTwdMzZ24

It suggest to use 
```
font: -apple-system-body;
```
I tried it through inspect element, and it seems to be working.

**Comment by jainkuniya** [comment]
I tried building iOS on https://github.com/zulip/zulip-mobile/commit/642b2d43bf4b2c3d79f385475aeae032e6f6a3fd, but facing some issue related to Sentry
![image](https://user-images.githubusercontent.com/18511177/80303637-10ea6c00-87cf-11ea-952a-02d74afe5d96.png)

**Comment by gnprice** [comment]
> I tried building iOS on [642b2d4](https://github.com/zulip/zulip-mobile/commit/642b2d43bf4b2c3d79f385475aeae032e6f6a3fd), but facing some issue related to Sentry

From your other comment, and #4071 (I had to go look that up -- reminder to please link to the issue from the PR as well as the main commit message), I think we may have a solution to the underlying issue here!

But I would still be interested in debugging so that you're able to build old versions, because that is a valuable investigative tool. A chat thread would be a good place. Please include details of what you did that led up to the error, and what you've tried so far to fix it.

**Comment by jainkuniya** [comment]
So I checked out  642b2d4
- did rm -rf `node_modules` and `yarn install`
- cleaned `ios/build`
- `npx react-native run-ios`
This gave me error 
```
Error: Could not find iPhone X simulator
```
Fixed it by https://github.com/facebook/react-native/issues/23282#issuecomment-476439080
- Now the iOS build is failing
```
error: Build input file cannot be found: '.........../zulip-mobile/node_modules/react-native/ReactCommon/yoga/yoga/YGValue.cpp'
```

- Then I opened xcode, got the same error
- Then I changed the build system to Legacy https://github.com/facebook/react-native/issues/23886#issuecomment-472378459
- still got the same error
- then as usual I deleted cache
```
➜  zulip-mobile git:(642b2d43) ✗ rm -rf ~/Library/Caches/CocoaPods
➜  zulip-mobile git:(642b2d43) ✗ rm -rf ios/build
➜  zulip-mobile git:(642b2d43) ✗ rm -rf ios/Pods
```
- now build from xcode succeeded 🎉 and from terminal as well

Not sure why other day I was  getting issue related to Sentry 🤔 (and day before this I was getting some issue related to Yoga)

But on iPhone X 12.4 and 12.1, this commit doesn't seems to work
![image](https://user-images.githubusercontent.com/18511177/80523840-5021e400-89ac-11ea-9bc9-b41be6f2dd15.png)
![image](https://user-images.githubusercontent.com/18511177/80525628-2b7b3b80-89af-11ea-8ce7-2a6b7d0a910a.png)
![image](https://user-images.githubusercontent.com/18511177/80525706-464db000-89af-11ea-9a85-fe609bb26c1d.png)
